pyOCD/gdbserver/gdbserver.py

```diff
@@ -1280,21 +1280,15 @@ def getThreadsXML(self):
         if not self.is_threading_enabled():
             t = SubElement(root, 'thread', id="1", core=str(self.core))
             if self.is_target_in_reset():
-                t.text = "Reset"
+                t.set('name', "Reset")
             else:
-                t.text = "Thread mode"
+                t.set('name', "Thread mode")
         else:
             threads = self.thread_provider.get_threads()
             for thread in threads:
                 hexId = "%x" % thread.unique_id
-                t = SubElement(root, 'thread', id=hexId, core="0", name=thread.name, handle=hexId)
-
-                desc = thread.description
-                if desc:
-                    desc = thread.name + "; " + desc
-                else:
-                    desc = thread.name
-                t.text = desc
+                t = SubElement(root, 'thread', id=hexId, name=thread.name)
```

Comment:
Add back in the core attribute and I'll be happy:

```suggestion
                t = SubElement(root, 'thread', id=hexId, name=thread.name, core=str(self.core))
```

While thread providers don't provide core information (SMP is not supported), the gdbserver knows the core to which it is attached. The actual core number should have been included, but that got missed somehow.

Reply:
Still seems like clutter to me. If not SMP, it's not per-thread info. Having it present makes Eclipse put "[core 0]" on every line, which is a bit like putting "[pid 720]" on every line. You should already know which CPU/process you're looking at from context or UI indication (image name/connection name/port number etc).

No GDB UI should be mixing threads from different processes/remotes so that you would have to distinguish them like this.